tests/test_ai/test_priority_algorithm/test_mutations/test_robinhood.py

```diff
@@ -114,19 +120,24 @@ def test_mutate_robinhood__does_mutate_when_not_optimal(self):
         The mutated team set should be different from the original team set.
         Not guaranteed to change, so this test may fail. (~(1/3)^100 = 2e-48 chance of failing)
         """
-        for mutate_func in [mutate_robinhood, mutate_robinhood_holistic]:
+        functions = [mutate_robinhood, mutate_robinhood_holistic]
+        for mutate_func in functions:
             priority_team_set, student_dict = create_new_priority_team_set(3, 9)
             priorities = [StudentListPriority([1, 2])]
-
+            result = False
             for _ in range(100):
+                random.shuffle(
+                    priority_team_set.priority_teams
+                )  # Randomize the order so that deterministic algorithms don't get stuck
                 mutated_team_set = mutate_func(
                     priority_team_set.clone(), priorities, student_dict
                 )
                 if not equal_priority_team_sets(priority_team_set, mutated_team_set):
-                    return
+                    result = True
+                    break
 
             self.assertTrue(
-                False,
+                result,
                 "The mutated team set should be different from the original team set. NOTE: This test may fail because the mutated team set is not guaranteed to change.",
```

Comment:
Oh wait why is this true again? You can merge this PR because it's unrelated, but make a ticket for me or you to make this test not "sometimes fail"

Flaky tests are the enemy